metrics/size.table/query.go

```diff
@@ -0,0 +1,55 @@
+package sizetable
+
+import (
+	"fmt"
+	"strings"
+)
+
+func TableSizeQuery(set map[string]string) (string, error) {
+	query := "SELECT table_schema AS db, table_name as tbl, COALESCE(data_length + index_length, 0) AS tbl_size_bytes FROM information_schema.TABLES"
+	where := " WHERE "
+	if include := set[OPT_INCLUDE]; include != "" {
+		var include_tables []string
+		if strings.Contains(set[OPT_INCLUDE], ",") {
+			include_tables = strings.Split(set[OPT_INCLUDE], ",")
+		} else {
+			include_tables = []string{set[OPT_INCLUDE]}
+		}
+		where = setWhere(include_tables, true, where)
+	} else {
+		where = where + "NOT "
+		var exclude_tables []string
+		if strings.Contains(set[OPT_EXCLUDE], ",") {
+			exclude_tables = strings.Split(set[OPT_EXCLUDE], ",")
+		} else {
+			exclude_tables = []string{set[OPT_EXCLUDE]}
+		}
+		where = setWhere(exclude_tables, false, where)
+	}
+	return query + where, nil
+}
+
+func setWhere(tables []string, isInclude bool, where string) string {
+	for i, exclude_table := range tables {
+		if strings.Contains(exclude_table, ".") {
+			dbAndTable := strings.Split(exclude_table, ".")
+			db := dbAndTable[0]
+			table := dbAndTable[1]
+			if table == "*" {
+				where = where + fmt.Sprintf("(table_schema = '%s')", db)
+			} else {
+				where = where + fmt.Sprintf("(table_schema = '%s' AND table_name = '%s')", db, table)
+			}
+		} else {
+			where = where + fmt.Sprintf("(table_name = '%s')", exclude_table)
+		}
+		if i != (len(tables) - 1) {
+			if isInclude {
+				where = where + " OR "
+			} else {
+				where = where + " AND NOT "
```

Comment:
Why,
```
SELECT table_schema AS db, table_name as tbl, COALESCE(data_length + index_length, 0) AS tbl_size_bytes FROM information_schema.TABLES WHERE NOT (table_schema = 'mysql') AND NOT (table_schema = 'information_schema') AND NOT (table_schema = 'performance_schema') AND NOT (table_schema = 'sys');
```

and not more simply,

```
SELECT table_schema AS db, table_name as tbl, COALESCE(data_length + index_length, 0) AS tbl_size_bytes FROM information_schema.TABLES WHERE table_schema NOT IN ('mysql','information_schema','performance_schema','sys');
```
? 

Granted, you'd still have to handle cases for `table_name [NOT] IN (...)`, too.  Point is: `IN()` is more compact and more common SQL.

Reply:
if we do:
`table_schema NOT IN ('my_sql', 'sys') and table_name NOT IN ('table_1', 'table_2')`

My worry is that it'll include/exclude tables and dbs which we don't intend on targeting. If our 'my_sql' and 'sys' dbs looked like:

sys: table_1
my_sql: table_1, table_2

and we entered an input like this to exclude table_1 from my_sql and table_2 from sys:
`my_sql.table_1,sys.table_2`

Using this formatting:
- foo = any table named food (in any db)
- foo.* = all tables in db foo
- foo.bar = table bar in db foo

the query would exclude table_1 from the sys db as well even when that wasn't the users intention.

Reply:
Right... my example was not a good one, but it's ok: we can start with this method since it works and table size collection (should be) low frequency, so performance isn't as critical.